166816212 Comment on articles

vincentayorinde opened the pull request:
#### Description
 This PR allows a user to be able to add a comment to articles 

#### Type of change
Please select the relevant option
- [ ] Bug fix (non-breaking changes which fixes an issue)
- [x] New feature (non-breaking changes which adds functionality)
- [ ] Breaking change (fix or feature that could cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Chore

#### How Has This Been Tested?
please describe the test that you ran to verify your changes

- [ ] End to end 
- [X] Integration
- [ ] Unit test

#### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] I have made corresponding changes to the documentation
- [x] My changes generates no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

#### Pivotal Tracker
[#166816212](https://www.pivotaltracker.com/story/show/166816212)

caleb-42 commented on controllers/comments/index.js:
```diff
@@ -0,0 +1,30 @@
+
+export default {
+  addComment: async (req, res) => {
+    const { params: { slug }, body: { content }, user } = req;
```
smart, nicely done

vincentayorinde replied:
Thank you for the feedback

caleb-42 commented on db/models/Comment.js:
```diff
@@ -0,0 +1,21 @@
+  Comment.associate = (models) => {
+    Comment.belongsTo(models.User, {
+      foreignKey: 'userId',
+      as: 'user'
```
can see you did not cascade the association of comment to users the way you did in the case of Articles on line 17, please can you explain why?

vincentayorinde replied:
Thank you for your feedback, I actually did, kindly expand the file for more detail.

caleb-42 commented on validators/articles/index.js:
```diff
@@ -92,5 +92,22 @@ export default {
+
+  addComment: async (req, res, next) => {
+    const rules = {
+      content: 'required|string'
```
I think there should also be a validation for the article slug coming from the request params object in case its not provided for or is not a string

vincentayorinde replied:
Hi @caleb-42, thank you for your feedback, I made use of the `articleSlug` validation which exists in the validations to check for the article slug coming from the request params object.

caleb-42 replied:
ok. nice

caleb-42 reviewed (commented):
Good job on the commenting feature for the articles. I await your response to my reviews

FrankChinedu commented on db/migrations/20190715134956-create-comment.js:
```diff
@@ -0,0 +1,42 @@
+    },
+    content: {
+      allowNull: false,
+      type: Sequelize.STRING
```
Nice PR, but I think you should use Sequelize.TEXT instead of Sequelize.STRING

vincentayorinde replied:
Good catch Frank, thank you for noticing, I found out while I debugging as well.

vincentayorinde committed:
166816212-feature(user, article, comment): comment on articles
- add comment role migration
- add comment controller
- add comments route
- add validation
- add tests
- add documentation
[Delivers #166816212]
